plugins/permission-node/api-report.md

```diff
@@ -115,17 +115,30 @@ export const createConditionTransformer: <
 export const createPermissionIntegrationRouter: <
   TResourceType extends string,
   TResource,
->(options: {
+>(
+  options: CreatePermissionIntegrationRouterOptions<TResourceType, TResource>,
+) => express.Router;
+
+// @public
+export type CreatePermissionIntegrationRouterOptions<
+  TResourceType extends string,
+  TResource,
+> =
+  | {
+      permissions: Array<Permission>;
+    }
+  | CreatePermissionIntegrationRouterResourceOptions<TResourceType, TResource>;
+
+// @public
+export type CreatePermissionIntegrationRouterResourceOptions<
+  TResourceType extends string,
+  TResource,
+> = {
   resourceType: TResourceType;
-  permissions?: Permission[] | undefined;
-  rules: PermissionRule<
-    TResource,
-    any,
-    NoInfer<TResourceType>,
-    PermissionRuleParams
-  >[];
-  getResources: (resourceRefs: string[]) => Promise<(TResource | undefined)[]>;
-}) => express.Router;
+  permissions?: Array<Permission>;
+  rules: PermissionRule<TResource, any, NoInfer<TResourceType>>[];
+  getResources?: GetResourcesFn<TResource>;
```

Comment:
Would you be alright with skipping the `GetResourcesFn` addition? I feel it mostly clutters the API surface. Otherwise we typically use `*Func` as the suffix, or even more commonly just give it a name like `ResourceFetcher`

Reply:
yep fine for us!